docs/core/diagnostics/metrics-instrumentation.md

```diff
@@ -103,33 +103,89 @@ to install it:
 > dotnet tool update -g dotnet-counters
 You can invoke the tool using the following command: dotnet-counters
-Tool 'dotnet-counters' (version '5.0.251802') was successfully installed.
-```
-
-While the example app is still running, list the running processes in a second shell to determine the process ID:
-
-```dotnetcli
-> dotnet-counters ps
-     10180 dotnet     C:\Program Files\dotnet\dotnet.exe
-     19964 metric-instr E:\temp\metric-instr\bin\Debug\netcoreapp3.1\metric-instr.exe
+Tool 'dotnet-counters' (version '7.0.430602') was successfully installed.
 ```
 
-Find the ID for the process name that matches the example app and have dotnet-counters monitor the new counter:
+While the example app is still running, use dotnet-counters to monitor the new counter:
 
 ```dotnetcli
-> dotnet-counters monitor -p 19964 HatCo.HatStore
+> dotnet-counters monitor -n metric-demo.exe --counters HatCo.Store
 Press p to pause, r to resume, q to quit.
     Status: Running
 
-[HatCo.HatStore]
-    hats-sold (Count / 1 sec)                          4
+[HatCo.Store]
+    hatco.store.hats_sold (Count / 1 sec)                          4
 ```
 
 As expected, you can see that HatCo store is steadily selling 4 hats each second.
 
+## Get a Meter via dependency injection
+
+In the previous example, the Meter was obtained by constructing it with `new` and assigning it to a static field. Using statics this way is not a good approach when using dependency
+injection (DI). In code that uses DI, such as ASP.NET Core or apps with [Generic Host](../extensions/generic-host.md), create the Meter object using
+<xref:System.Diagnostics.Metrics.IMeterFactory>. Starting in .NET 8, hosts will automatically register <xref:System.Diagnostics.Metrics.IMeterFactory> in the service container
+or you can manually register the type in any <xref:Microsoft.Extensions.DependencyInjection.IServiceCollection> by calling <xref:Microsoft.Extensions.DependencyInjection.MetricsServiceExtensions.AddMetrics%2A>.
+The meter factory integrates metrics with DI, keeping Meters in different service collections isolated from each other even if they use an identical name. This is
+especially useful for testing so that multiple tests running in parallel only observe measurements produced from within the same test case.
+
+To obtain a Meter in a type designed for DI, add an <xref:System.Diagnostics.Metrics.IMeterFactory> parameter to the constructor, then call
+<xref:System.Diagnostics.Metrics.MeterFactoryExtensions.Create%2A>. This example shows adding IMeterFactory to an ASP.NET Core MVC controller type:
+
+```cs
+public class HatStoreController : Controller
+{
+    private readonly Counter<int> _hatsSold;
+
+    public HatStoreController(IMeterFactory meterFactory)
+    {
+        var meter = meterFactory.Create("HatCo.Store");
+        _hatsSold = meter.CreateCounter<int>("hatco.store.hats_sold");
+        // Other methods in this class can now use the _hatsSold counter to increment the count.
+    }
```

Comment:
I'm not sure we'd want to suggest this as an option at all. A new `Controller` instance is created for every request.

I know Meter and counters are cached, so it doesn't matter too much, but it feels clean to "create" these types once in their own type like is suggested immediately below.

Reply:
Removed